src/_arraykit.c

```diff
@@ -383,7 +383,28 @@ dtype_from_element(PyObject *Py_UNUSED(m), PyObject *arg)
 
     // Integers
     if (PyLong_CheckExact(arg)) {
-        return (PyObject*)PyArray_DescrFromType(NPY_LONG);
+        int overflow;
+        npy_int64 v = PyLong_AsLongLongAndOverflow(arg, &overflow);
+        if (v != -1 && overflow == 0) {
+            // Prioritize fastest case first
```

Comment:
You need to check this first, otherwise I think `-1` will return `object` (maybe include that as a test case). It shouldn't have a noticeable speed difference, since checking if `v == -1` is basically free and very rarely succeeds:

```suggestion
        if (v == -1 && PyErr_Occurred()) {
            return NULL;
        }
        if (overflow == 0) {
```